service/create/policy.go

```diff
@@ -102,3 +116,69 @@ func createInstanceProfile(svc *iam.IAM, clusterID string) (string, error) {
 
 	return clusterProfileName, nil
 }
+
+func deleteInstanceProfile(svc *iam.IAM, clusterID string) error {
+	clusterProfileName := fmt.Sprintf("%s-%s", clusterID, ProfileNameTemplate)
+
+	_, err := svc.DeleteInstanceProfile(&iam.DeleteInstanceProfileInput{
+		InstanceProfileName: aws.String(clusterProfileName),
+	})
+
+	if err != nil {
+		return err
```

Comment:
We want to use error masking. See for instance https://github.com/giantswarm/kubernetesd/blob/ede82bd2f94dbc11cefc95349652b29b6dc41b44/service/service.go#L75.